src/connectRoutes.js

```diff
@@ -126,6 +130,14 @@ export default (
     initialDispatch: shouldPerformInitialDispatch = true
   }: Options = options
 
+  const selectLocationState = typeof locationKey === 'function'
+    ? locationKey
+    : state => state[locationKey]
+
+  const selectTitleState = typeof titleKey === 'function'
+    ? titleKey
+    : state => state[titleKey]
+
```

Comment:
This is where the overloading happens. I'm not a huge fan of overloading like this (overloading in general reminds me of jQuery... 😱) but I understand the desire not to change the API yet.